### ISSUE TITLE
Application config not set as environment variables 

### ISSUE THREAD

**Issue by Segflow** [Azure/Azure-Functions#1696]
The [documentation says](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#environment-variables) that application settings are exposed as environment variables during execution.

I have an Azure function app setup (version 3), I configured few application settings. 

Within that app I created a new Nodejs Azure function, and I can't see any my of application setting when printing/accessing them. 

The trigger of my function is EventHub, and I when logging `process.env` I see 
```
SUDO_COMMAND:
   '/azure-functions-host/Microsoft.Azure.WebJobs.Script.WebHost',
  PATH:
   '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
  TERM: 'xterm',
  HOME: '/home',
  SHELL: '/bin/bash',
  SUDO_GID: '0',
  languageWorkers__python__defaultRuntimeVersion: '3.6',
  Fabric_NodeIPOrFQDN: '10.92.0.9',
  JAVA_HOME: '/usr/lib/jvm/zre-8-azure-amd64',
  Fabric_NetworkingMode: 'Other',
  SUDO_UID: '0',
  MESH_INIT_URI: 'http://localhost:6060/',
  CONTAINER_NAME: '33210E11-637335464109795532',
  Fabric_ReplicaId: '132424232167917561',
  Fabric_CodePackageName: 'functionsRuntimeContainer',
  Fabric_ServiceName: 'functionsRuntimeService',
  ServiceName: 'functionsRuntimeService',
  WEBSITE_CLOUD_NAME: 'Azure',
  ASPNETCORE_VERSION: '3.0.0',
  Location: 'East US',
  ResourceType: 'Microsoft.ServiceFabricMesh/applications',
  CONTAINER_START_CONTEXT_SAS_URI:
   'http://wawsstorageprodblu089.blob.core.windows.net/...',
  DOTNET_RUNNING_IN_CONTAINER: 'true',
  ASPNETCORE_URLS: 'http://localhost:9091',
  LANG: 'C.UTF-8',
  WEBSITE_STAMP_DEPLOYMENT_ID: 'b30d3459837e4ad386606b91c7cb0da4',
  AzureWebJobsScriptRoot: '/home/site/wwwroot',
  WEBSITE_HOME_STAMPNAME: 'waws-prod-blu-089',
  CONTAINER_ENCRYPTION_KEY: 'AM1tVZ23U1Oi4E3LNsjIzkdwwySzdnSaUEYU+xiN9mM=',
  Fabric_ApplicationName: 'App-33210E11-637335464109795532',
  ResourceName: 'App-33210E11-637335464109795532',
  Fabric_ServiceDnsName: 'functionsRuntimeService.App-33210E11-637335464109795532',
  Fabric_Epoch: '132424232148856091:8589934592',
  DOTNET_USE_POLLING_FILE_WATCHER: 'true',
  ResourceGroupName: 'AZCONTAINERS-WAWS-PROD-BLU-089-1',
  SubscriptionId: '<REDACTED>',
  PWD: '/',
  Fabric_Id: '731beced-ce7d-4d33-b3d3-34ee48861484',
  CodePackageName: 'functionsRuntimeContainer',
  HOST_VERSION: '3.0.14287.0',
  Fabric_ReplicaName: '0',
  LOGNAME: 'app',
  HOSTNAME: 'SandboxHost-637335464132842685',
  WEBSITE_PLACEHOLDER_MODE: '1',
  USER: 'app',
  CONTAINER_IMAGE_URL: 'mcr.microsoft.com/azure-functions/mesh:3.0.14287',
  SUDO_USER: 'root',
  FUNCTIONS_WORKER_DIRECTORY: '/azure-functions-host/workers/node'
```

I don't see any of the application config there. 

<img width="692" alt="Screen Shot 2020-08-20 at 8 21 38 PM" src="https://user-images.githubusercontent.com/2049730/90815553-c9aa5d00-e322-11ea-95ee-ba3885ba62f7.png">

**Comment by Fuco1** [comment]
I'm having the same issue, my application settings are not visible in the environment of the function runtime.

**Comment by Segflow** [comment]
Hey @Fuco1, are you also creating the azure function using terraform? if so you need to explicitly set this app settings:
`FUNCTIONS_WORKER_RUNTIME` and set it's value to `node`

**Comment by Segflow** [comment]
I remember I asked azure support to add this to their documentation, seems like they did not.

**Comment by Fuco1** [comment]
@Segflow yes indeed, I first created the function through the UI and when I was happy with the settings I re-created it with terraform, and that's when it broke.  I'll try to add the setting you mentioned and see what happens.  Thanks for the tip!

**Comment by ludflu** [comment]
I just spent a couple hours trying to debug this. this is a serious bug and I can't believe its been open since August 2020.

**Comment by Segflow** [comment]
> I just spent a couple hours trying to debug this. this is a serious bug and I can't believe its been open since August 2020.
> 
> The same issue occurs if you try to deploy a python function app using terraform.

you need to explicitly add this app settings:
`FUNCTIONS_WORKER_RUNTIME` and set its value to `python`

**Comment by ludflu** [comment]
yup, that's what I did. works fine now.

**Comment by v-bbalaiagar** [comment]
Hi @Segflow, Apologies for the delayed response, Were you able to find a solution here. Kindly let us know if you could resolve it can we close this issue?

**Comment by v-bbalaiagar** [comment]
Hi @Segflow, Apologies for the confusion

@ggailey777, Tagging gailey, Since its a document related issue. Can we add this in our documents and if its a valid note, then can we transfer this issue to the docs repo.

**Comment by REslim30** [comment]
I'm running into this same issue. Terraform + function apps (in js) = no environment variables.

When I add in this app setting:
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/43259657/189273643-a75e4c7d-b9ad-43a0-8f1d-b14944fad89d.png">

I get 503 responses (http triggered function)
<img width="1108" alt="image" src="https://user-images.githubusercontent.com/43259657/189273702-673a0d0f-ea9f-4872-8282-234b5f9b5fa5.png">

And I also get this warning:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/43259657/189273781-7217e8a2-ccde-40cf-8ee5-a151f1547121.png">

When I remove `FUNCTIONS_WORKER_RUNTIME`, my functions operate normally (but no environment variables)...

**Comment by REslim30** [comment]
Ok the solution was to set `FUNCTIONS_EXTENSION_VERSION` to `~3` (mine was on `~4` before) and also `FUNCTIONS_WORKER_RUNTIME` to `node`. Not sure why this works or what the implications are.

Thank you @Segflow, for the fix. I spent about 4.5 hrs debugging this issue...

**Comment by REslim30** [comment]
@v-bbalaiagar is there a fix for this issue?

Setting `FUNCTIONS_EXTENSION_VERSION` to 3 is no longer a viable option as support is ending on 30 September 2022.

<img width="574" alt="image" src="https://user-images.githubusercontent.com/43259657/191871655-de528a44-ccb6-4fc3-b78c-b33455374182.png">

If not, are there any other workarounds? Otherwise, could you direct me where I can file an issue?

**Comment by REslim30** [comment]
Ok so the solution for this was to set the `node_version` in `application_stack` in `site_config` instead of `FUNCTIONS_WORKER_RUNTIME`. This allows you to set node 16 (which runs on runtime version 4.x).
